packages/fx-core/src/core/middleware/projectMigrator.ts

```diff
@@ -269,7 +261,9 @@ async function migrateToArmAndMultiEnv(ctx: CoreHookContext): Promise<void> {
     telemetryProperties
   );
 
-  await preCheckKeyFiles(projectPath, ctx);
+  if (!(await preCheckKeyFiles(projectPath, ctx))) {
```

Comment:
Is it better to put this check in `needMigrateToArmAndMultiEnv()`? If the key project files are broken, do not even pop up migrate dialog?

Reply:
preCheckKeyFiles is part of `migrateToArmAndMultiEnv`, I think we shouldn't check key files if user doesn't want to upgrade.

Reply:
Then add a telemetry event if preCheckKeyFiles fails. Otherwise, we will lose this info (previously, errorHandlerMW can catch this exception and send telemetry).